interface/main/tabs/main.php

```diff
@@ -510,5 +509,31 @@ function jsFetchGlobals(scope) {
 document.addEventListener('touchstart', {}); //specifically added for iOS devices, especially in iframes
 </script>
 
+<script>
+    // set up global translations for js
+    function setupI18n(lang_id) {
+        top.restoreSession();
+        fetch(<?php echo js_escape($GLOBALS['webroot'])?> + "/library/ajax/i18n_generator.php?lang_id=" + encodeURIComponent(lang_id), {
+            credentials: 'same-origin',
+            method: 'GET'
+        })
+        .then(response => response.json())
+        .then(data => {
+            var translationObj = data;
+            i18next.init({
+                lng: 'selected',
+                debug: false,
+                resources: {
+                    selected: {
+                        translation: translationObj
+                    }
+                }
+            });
+        })
```

Comment:
Maybe we can add new JS function -
```
function xl(string) {
   top.i18next.t(string)\
}
```

Reply:
> We need check that ajax with translations return

That's true for any JS scripts running in main but once main.php loads, xlate are available elsewhere.
I like idea of xl wrapper.